### PR TITLE
style(route) use"controller as"

### PR DESCRIPTION
**Pull request by remicastaing**
- use "controller as"
- apply JPSG

### CODE REVIEW

File: templates/component/name.component.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ angular.module('<%= scriptAppName %>')
     template: '<h1>Hello {{ $ctrl.message }}</h1>',
     bindings: { message: '<' },
     controller: function() {
-      this.message = 'World'
+      var vm = this;
+      vm.message = 'World';
```
Comment (Awk34) [inline templates/component/name.component.js]: This seems unnecessary, don't you think?

Comment (remicastaing) [inline templates/component/name.component.js]: I just did as for all other component. But, perhaps, it's unnecessary. There is no styleguide for component anywhere.

Comment (Awk34) [inline templates/component/name.component.js]: I'd just keep the `this.message = 'World';`. No need to create a new reference to `this.`

File: templates/route/name.controller.js [hunk 2]
```diff
@@ -6,14 +6,16 @@ class <%= classedName %>Component {
     this.message = 'Hello';
   }
 }<% } else { %>
-function <%= classedName %>Component($scope) {
-  $scope.message = 'Hello';
+function <%= classedName %>Component() {
+  var vm = this;
+  vm.message = 'Hello';
```
Comment (Awk34) [inline templates/route/name.controller.js]: same here, just put on `this`

File: templates/component/name.component.js (hunk 1 above)
Comment (remicastaing) [inline templates/component/name.component.js]: In this Post : https://johnpapa.net/angularjss-controller-as-and-the-vm-variable/, there is some explanations for var vm = this. It seems also to apply in the component case.
But I would create a named function and not just inlining the function inside the component.
What do you think?

Comment (Awk34) [inline templates/component/name.component.js]: not inlining the controller function is good, but creating a new `vm = this` here is bad IMO. It's just unnecessarily turning a 1-line controller into a 2-line controller. Now, if your controller was much larger it could make sense to store  the class `this` in a var, but not here.

File: templates/controller/name.controller.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 'use strict';
 
+function <%= cameledName %>Controller() {
+  var vm = this;
+  vm.message = 'Hello';
```
Comment (Awk34) [inline templates/controller/name.controller.js]: ^

File: templates/controller/name.controller.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
+}
+
 angular.module('<%= scriptAppName %>')
-  .controller('<%= classedName %>Ctrl', function ($scope) {
-    $scope.message = 'Hello';
-  });
+  .controller('<%= classedName %>Ctrl', <%= cameledName %>Controller);
```
Comment (Awk34) [inline templates/controller/name.controller.js]: While you're at it, want to rename to `'<%= classedName %>Controller'`?

File: templates/decorator/name.decorator.js [hunk 5]
```diff
@@ -1,9 +1,13 @@
...
+}
+
 angular.module('<%= scriptAppName %>')
-  .config(function ($provide) {
-    $provide.decorator('<%= cameledName %>', function ($delegate) {
-      // decorate the $delegate
-      return $delegate;
-    });
-  });
+  .config(<%= cameledName %>DecoratorConfig);
```
Comment (Awk34) [inline templates/decorator/name.decorator.js]: I think we can move the Decorator to a named function, but I think `<%= cameledName %>DecoratorConfig` is unnecessary, just inline it

File: templates/route/name.html [hunk 6]
```diff
@@ -1 +1 @@
-<div>This is the <%= name %> view.</div>
+<div>This is the <%= name %> view. And the message: {{<%= classedName %>.message}}</div>
```
Comment (Awk34) [inline templates/route/name.html]: This should be in its own commit
